### PR TITLE
Rewrite OpenShift docs for 3.9

### PR DESCRIPTION
**Pull request by oicheryl**

### CODE REVIEW

File: _docs/install/kubernetes/dynamic-provisioning.md [hunk 1]
```diff
@@ -9,11 +9,12 @@ module: install/kubernetes/dynamic-provisioning
 
 StorageOS volumes can be created on-demand through dynamic provisioning.
 
-1. Adminstrators create storage classes to define different types of storage,
-using a secret to authenticate against the StorageOS API.
-1. Users create a persistent volume claim and references it in a pod.
+1. Create the secret needed to authenticate against the StorageOS API.
+1. Adminstrators create storage classes to define different types of storage.
```
Comment (croomes) [inline _docs/install/kubernetes/dynamic-provisioning.md]: The api secret and storage class are created by the Helm chart, so this only applies if using the container install method.  I'd move these to the container install method doc?

File: _docs/install/kubernetes/dynamic-provisioning.md [hunk 2]
```diff
@@ -31,14 +32,14 @@ changed its password (recommended), you must specify the new settings in
 `apiAddress`, `apiUsername` and `apiPassword`, encoded as base64 strings.
 
 ```bash
-$ echo -n "tcp://127.0.0.1:5705" | base64
+echo -n "tcp://127.0.0.1:5705" | base64
```
Comment (croomes) [inline _docs/install/kubernetes/dynamic-provisioning.md]: maybe include a note here to explain that the k8s master needs to be able to access the api, so if the master doesn't run the storageos container then a LoadBalancer ip may be needed.  (I can give a sample yaml if needed)

File: _docs/install/openshift/dynamic-provisioning.md [hunk 3]
```diff
@@ -0,0 +1,282 @@
...
+anchor: install
+module: install/openshift/dynamic-provisioning
+---
+
+# Dynamic Provisioning
+
+StorageOS volumes can be created on-demand through dynamic provisioning.
+
+1. Create the secret needed to authenticate against the StorageOS API.
+1. Adminstrators create storage classes to define different types of storage.
```
Comment (croomes) [inline _docs/install/openshift/dynamic-provisioning.md]: same as k8s comment, Helm will create the secret and storage class

File: _docs/install/openshift/preprovisioned.md [hunk 4]
```diff
@@ -0,0 +1,169 @@
...
+     storageos:
+       # This volume must already exist within StorageOS
+       volumeName: nginx-pv01
+       # volumeNamespace is optional, and specifies the volume scope within
+       # StorageOS.  Set to `default` or leave blank if you are not using
+       # namespaces.
+       volumeNamespace: default
+       # The filesystem type to create on the volume, if required.
+       fsType: ext4
+   ...
```
Comment (croomes) [inline _docs/install/openshift/preprovisioned.md]: ...?  Is that a yaml thing?

File: _docs/install/openshift/index.md [hunk 5]
```diff
@@ -0,0 +1,58 @@
...
+
+1. If the kubelet is running in a container, add `--volume=/var/lib/storageos:/var/lib/storageos:rshared` to each
+of the kubelets. This will be [fixed](https://github.com/kubernetes/kubernetes/pull/58816) in OpenShift 3.10+.
+
+1. The [iptables rules]({% link
+_docs/install/prerequisites/firewalls.md %}) required for StorageOS.
+
+1. [Enable the Network Block Device module]({% link
+_docs/install/prerequisites/devicepresentation.md %}) on each node for better
+performance.
```
[the remark this answers is not in the document]
Comment (oicheryl) [reply _docs/install/openshift/index.md]: Done.

File: _docs/install/kubernetes/dynamic-provisioning.md (hunk 2 above)
Comment (oicheryl) [reply _docs/install/kubernetes/dynamic-provisioning.md]: Added note. Yes please to the sample yaml.

File: _docs/install/openshift/preprovisioned.md (hunk 4 above)
Comment (oicheryl) [reply _docs/install/openshift/preprovisioned.md]: Not sure - copied from the existing docs. @senax?

File: _docs/install/kubernetes/dynamic-provisioning.md (hunk 1 above)
Comment (oicheryl) [reply _docs/install/kubernetes/dynamic-provisioning.md]: I've added a note to skip to step 3 if you used the helm chart.

File: _docs/install/openshift/dynamic-provisioning.md (hunk 3 above)
Comment (oicheryl) [reply _docs/install/openshift/dynamic-provisioning.md]: I've added a note to skip to step 3 if you used the helm chart.

**Pull request merged**